Remove stropts.h?

I do not have any personal involvement with this particular bug, I just read about it on the forums [today](https://forum.mysensors.org/topic/11246/why-is-stropts-h-included/2) where mfalkvidd helpfully linked to [an earlier thread about the same issue](https://forum.mysensors.org/topic/11138/mysensors-build-problem-on-ubuntu-20-04-rpi-or-tinker-board) wherein Yveaux added [a link to Stack Overflow](https://stackoverflow.com/questions/61029226/how-do-i-install-libraries-for-stropts-h) apparently explaining that this `stropts.h` has been deprecated?

I searched the issues here for "stropts" and did not find anything.  And since that earlier thread was back on 2020-05-06, and the OP apparently did not take Yveaux's hint to file a bug here, I am taking it upon myself to do so.

FWIW, I have also [begun](https://forum.mysensors.org/post/106458) trying to "teach people how to fish" and suggest they can help the project a lot by filing bug reports here.  In my case however I decided to create an issue in my own Orgmode file locally with a followup scheduled in about a week, in case the OP does not take it upon themselves to do so, hopefully I will remember to this way.

To be perfectly honest, even my own Orgmode / todo.txt systems are not the best, I often put things in there and then still somehow manage to forget about them altogether until some years later :smile:, so maybe eventually I just start immediately filing bug reports here before they have a chance to slip through the cracks.  I do try very hard though first to make people realize they can contribute easily, and encourage the idea that "many hands make light work" however everyone just seems so goddamn busy nowadays.

Cheers, and thanks guys for all your efforts over past years.  :beers:  It's the least I can do try and help out a little bit.

It seems that I have the same issue when compiling.
```
hal/architecture/Linux/drivers/core/interrupt.cpp:33:10: fatal error: stropts.h: No such file or directory
   33 | #include <stropts.h>
      |          ^~~~~~~~~~~
compilation terminated.
make: *** [Makefile:99: build/hal/architecture/Linux/drivers/core/interrupt.o] Error 1
```

It looks like removing the include resolves the issue, but I don't know if it affects the functionality in any way.

--- a/hal/architecture/Linux/drivers/core/interrupt.cpp
+++ b/hal/architecture/Linux/drivers/core/interrupt.cpp
@@ -30,7 +30,6 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
-#include <stropts.h>
 #include <errno.h>
 #include <sched.h>
 #include "log.h"

